src/Symfony/Component/Form/FormConfigBuilder.php

```diff
@@ -799,4 +820,16 @@ public static function isValidName(?string $name): bool
     {
         return '' === $name || null === $name || preg_match('/^[a-zA-Z0-9_][a-zA-Z0-9_\-:]*$/D', $name);
     }
+
+    /**
+     * @internal
+     */
+    public static function isEmpty($modelData): bool
```

Comment:
This should be defined as a default value for a new `is_empty` option (not sure about the name maybe `is_empty_callback`), so the FormType should be the only type responsible for calling `setIsEmptyCallback` by passing the option.
Then the `CheckboxType` will only redefine the option.